compiler/bindings/c/iree/compiler/dialects/iree_gpu.h

```diff
@@ -148,14 +155,26 @@ struct ireeGPUTargetInfo {
   MlirIdentifier arch;                // E.g., "gfx942".
   MlirAttribute subgroupSizeChoices;  // Subgroup size choices.
   MlirAttribute maxWorkgroupSizes;    // Max threads per X/Y/Z dimension.
-  int64_t maxThreadCountPerWorkgroup; // Max threads per workgroup.
-  int64_t maxWorkgroupMemoryBytes;    // Max workgroup memory.
+  int32_t maxThreadCountPerWorkgroup; // Max threads per workgroup.
+  int32_t maxWorkgroupMemoryBytes;    // Max workgroup memory.
+  MlirAttribute mmaIntrinsics;        // MMA Intrinsics.
 };
 
 // Queries GPU target info from the given `ExecutableTargetAttr` attribute.
 MLIR_CAPI_EXPORTED ireeGPUTargetInfo
 ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr);
 
+MLIR_CAPI_EXPORTED ireeGPUTargetInfo ireeGPUTargetInfoGet(
+    MlirContext mlirCtx, const char *arch, const int32_t *subgroupChoices,
+    size_t numSubgroupChoices, const int32_t *workgroupSizes,
+    size_t numWorkgroupSizes, int32_t threadCount, int32_t memoryBytes,
+    const mma_intrinsic_enum_t *mmaIntrinsics, size_t numMmaIntrinsics);
+
+MLIR_CAPI_EXPORTED void
+ireeGPUTargetInfoGetMMAIntrinsics(MlirAttribute mmaIntrinsics,
+                                  mma_intrinsic_enum_t *mmaIntrinsicVals,
+                                  uint8_t *isVirtuals, size_t numElements);
```

Comment:
I don't think we need to pass in the number of elements -- this can be derived from the size of the `mmaIntrinsics` array, no?